fiksi/src/lib.rs

```diff
@@ -149,14 +147,56 @@ pub enum ElementValue {
     Circle(kurbo::Circle),
 }
 
+/// The decomposer used for splitting the full geometric constraint system into smaller, solvable
+/// subsystems.
+#[derive(Clone, Debug, PartialEq)]
+pub enum Decomposer {
+    /// No decomposer. Solve the entire system at once using a numeric optimizer.
+    ///
+    /// This is generally stable for small systems, and is the fastest method for very small
+    /// systems on the order of tens of elements. Solving without decomposition becomes intractable
+    /// as systems grow beyond hundreds of elements.
+    None,
+
+    /// Decompose the system of non-linear equations by matching constraints' expressions to
+    /// elements' variables.
+    ///
+    /// This decomposition is relatively cheap to compute, and results in an ordering of groups of
+    /// constraints' residual expressions and the elements' variables they calculate. Each
+    /// expression calculates one variable. A single pass through these partially ordered groups,
+    /// solving for all constraints within each group together, solves the system.
+    ///
+    /// This works well for fully-determined systems, especially those fully anchored to the global
+    /// coordinate system; this works less well for systems with under-constrained parts. Severely
+    /// under-constrained systems tend to generate big expression groups, which requires many
+    /// expressions to be solved for at once.
+    SinglePass,
+
+    /// Warning: experimental. Perform a geometric recursive assembly of the system.
+    ///
+    /// This currently does not correctly handle constraints relative to the global coordinate
+    /// system, nor fixing elements to the global coordinate system.
+    ///
+    /// This searches for minimal, rigid clusters of elements and constraints within the geometric
+    /// system that can be solved separately. These rigid clusters can be transformed through
+    /// displacement relative to each other, but internal geometry remains unchanged. A partial
+    /// order for solving is induced by the hierarchy of such clusters.
+    ///
+    /// This works well for rigidly well-constrained systems, and is relatively robust against
+    /// under-constrained geometry. All rigidly well-constrained geometry will be found and solved
+    /// individually. Any left-over under-constrained part still requires solving for all its
+    /// expressions at once.
+    RecursiveAssembly,
+}
```

Comment:
We may want these or the numeric solver options (Levenberg-Marquardt/L-BFGS) to become parameterized in the future, to allow some more control.